lib/src/kvs/ds.rs

```diff
@@ -193,15 +197,29 @@ impl Datastore {
 	/// # }
 	/// ```
 	pub async fn new(path: &str) -> Result<Datastore, Error> {
+		Self::new_full(path, None).await
+	}
+
+	// For testing
+	#[allow(dead_code)]
+	pub async fn new_full(
```

Comment:
Ideally these testing types should be in a `tests` module or something like that that's gated by `#[cfg(test)]` so they don't leak as public API or be seen as such by tools like `cargo-semver-checks`.